### PR TITLE
Vulkan suboptimal swapchain fixes

### PR DESCRIPTION
**Pull request by Plagman**

### CODE REVIEW

File: rpcs3/main.cpp [hunk 1]
```diff
@@ -118,6 +118,8 @@ int main(int argc, char** argv)
 	rlim.rlim_max = 4096;
 	if (::setrlimit(RLIMIT_NOFILE, &rlim) != 0)
 		std::fprintf(stderr, "Failed to set max open file limit (4096).");
+	// Work around crash on startup on KDE: https://bugs.kde.org/show_bug.cgi?id=401637
```
Comment (RipleyTom) [inline rpcs3/main.cpp]: You left the workaround from your other PR in this one.

Comment (Plagman) [reply rpcs3/main.cpp]: woop

Comment (Plagman) [reply rpcs3/main.cpp]: Hopefully I did it right this time, thanks!

**Comment by Plagman** [comment]
Thanks for looking at it. This triggers for radv on an X11 Linux desktop on
my side, if Freesync is enabled (compositing off). The rendering would die
if I was playing and got a notification, or if alt-tabbed away from the
window. With this patch applied locally I played for an hour, had the game
in the background, etc, without seeing the problem.

**Comment by kd-11** [comment]
That explains it then, I don't have a freesync monitor so I never saw it. I'll merge as a hotfix.

**Comment by Plagman** [comment]
Apparently it isn't Freesync specific, just transitioning between pageflipping and not (and pageflipping is required for Freesync). So you wouldn't ever see it on a desktop with compositing enabled.

**Pull request merged**